pkg/shared/utils/utils.go

```diff
@@ -53,6 +58,39 @@ func RunCommand(cmd *exec.Cmd) ([]byte, error) {
 	return outb.Bytes(), nil
 }
 
+func RunCommandAndParseOutputLineByLine(cmd *exec.Cmd, pfn parserFn, resChan chan any, logger *logrus.Entry) error {
+	// Get a pipe to read from standard out
+	r, err := cmd.StdoutPipe()
+	if err != nil {
+		return fmt.Errorf("failed to get command pipe: %w", err)
+	}
+
+	scanner := bufio.NewScanner(r)
+	// Use the scanner to scan the output line by line and parse it
+	done := make(chan struct{})
+	go func() {
+		// Read line by line and process it
+		for scanner.Scan() {
+			line := scanner.Text()
+			resChan <- pfn(line, logger)
+		}
+		done <- struct{}{}
```

Comment:
Probably better to `close(done)` instead of putting a object on the channel to indicate doneness. Even better to do it with a defer to make sure that the code doesn't deadlock if there is a panic:

```
go func() {
   defer close(done)
   ...
}
```